buildenv/jenkins/common/variables-functions.groovy

```diff
@@ -730,10 +767,18 @@ def set_artifactory_config(id="Nightly") {
 
     if (VARIABLES.artifactory.defaultGeo) {
         // Allow default geo to be overridden with a param. Used by the Clenaup script to target a specific server.
-        ARTIFACTORY_CONFIG['defaultGeo'] = (params.ARTIFACTORY_GEO) ? params.ARTIFACTORY_GEO : VARIABLES.artifactory.defaultGeo
+        ARTIFACTORY_CONFIG['defaultGeo'] = params.ARTIFACTORY_GEO ?: VARIABLES.artifactory.defaultGeo
         ARTIFACTORY_CONFIG['geos'] = VARIABLES.artifactory.server.keySet()
-        ARTIFACTORY_CONFIG['repo'] = VARIABLES.artifactory.repo
-        ARTIFACTORY_CONFIG['uploadDir'] = "${ARTIFACTORY_CONFIG['repo']}/${JOB_NAME}/${BUILD_ID}/"
+        ARTIFACTORY_CONFIG['repo'] = get_value(VARIABLES.artifactory.repo, id) ?: get_value(VARIABLES.artifactory.repo, 'default')
+        ARTIFACTORY_CONFIG['uploadDir'] = get_value(VARIABLES.artifactory.uploadDir, id) ?: "${ARTIFACTORY_CONFIG['repo']}/${JOB_NAME}/${BUILD_ID}/"
```

Comment:
Shouldn't also define a default value for `uploadDir `(`get_value(VARIABLES.artifactory.uploadDir, default))`) in the variable files?
 e.g 
```
artifactory:
  uploadDir
    default: ${ARTIFACTORY_CONFIG['repo']}/${JOB_NAME}/${BUILD_ID}/
    nightly: ...
    release: ... 
```

Reply:
We could. The default is whatever is after the `?:`. Since our use cases don't require a different default I left it here.

Reply:
Artifactory settings are defined in the variables file. To be consistent, the uploadDir should also be defined in the varaibles file.